Frameworks/Ajax/Ajax/Sources/er/ajax/AjaxFileUploadRequestHandler.java

```diff
@@ -121,14 +121,9 @@ else if (formData.isFileUpload()) {
 							progress.dispose();
 							throw e;
 						}
-						FileOutputStream fos = new FileOutputStream(progress.tempFile());
-						try {
+						try (FileOutputStream fos = new FileOutputStream(progress.tempFile())) {
 							progress.copyAndTrack(uploadInputStream, fos, _maxUploadSize);
 						}
-						finally {
-							fos.flush();
```

Comment:
To be honest, the try-with-resources construct had passed me by until your commits here, so I'm no expert. It would seem, though, that it only guarantees `close()` will be called, and whether `close()` calls `flush()` is class-dependent. I'm not seeing a mention of `close()` calling `flush()` in the Javadocs for `FileOutputStream` or `OutputStream`. So, (a) do you know if it does, and (b) do you think it matters if it doesn't, or should we explicitly call `flush()` before the try-block ends?